### ISSUE TITLE
FilePath Documentation Patch

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5027]
|<img alt="KB1PKL's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| KB1PKL reported|
|-|-|
|Trac ID|trac#5027|
|Type|enhancement|
|Created|2011-04-05 19:36:02Z|
|Branch|https://github.com/twisted/twisted/tree/filepath-docstrings-5027|

Filled out some docstrings in `twisted.python.filepath.FilePath`, addressed ticket [#4736](https://github.com/twisted/twisted/issues/4736). Didn't modify anything (besides adding short blurb about unicode in `FilePath.__doc__`), just added.

Attachments:

* [filepath-api-docs.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/67b/67b1a89f60c5821810d9e832a9eed9d1b1f889ec/829f3c8d365f5c1d105316f7e0413e3fcb75ab1c.patch) (4280 bytes) - added by KB1PKL on 2011-04-05 19:36:34Z - Diff of my changes.
* [filepath-api-docs.2.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/67b/67b1a89f60c5821810d9e832a9eed9d1b1f889ec/4e59ecb33a9504728d7ba42ca7a89c9b364a649e.patch) (5058 bytes) - added by KB1PKL on 2011-04-06 19:19:14Z - Fixed issues with the new documentation.

<details><summary>Searchable metadata</summary>

```
trac-id__5027 5027
type__enhancement enhancement
reporter__KB1PKL KB1PKL
priority__normal normal
milestone__ 
branch__branches_filepath_docstrings_5027 branches/filepath-docstrings-5027
branch_author__exarkun__KB1PKL exarkun, KB1PKL
status__closed closed
resolution__fixed fixed
component__core core
keywords__filepath_documentation filepath documentation
time__1302032162000000 1302032162000000
changetime__1302582501000000 1302582501000000
version__None None
owner__ 
cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_2"></a>|@thijstriemstra set owner to KB1PKL|
|-|-|

Thanks for your patch. Couple of things:

 1. `C{FilePath}` should be `L{FilePath}`
 1. docstrings should be on 3 lines instead of 1

**Comment by twisted-trac** [comment]
|<img alt="KB1PKL's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_3"></a>|KB1PKL removed owner|
|-|-|

Fixed the issues (C{ should be L{, docstrings on three lines).

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun commented|
|-|-|

(In [31646]) Branching to 'filepath-docstrings-5027'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun commented|
|-|-|

Thanks for the patch KB1PKL.

  1. Don't ever use tabs for indentation!  Only use spaces.
  1. I don't think _Initialize the L{FilePath}_ tells the reader much that they didn't already know.  More useful would be something like _Convert a path string to an absolute path if necessary and initialize the L{FilePath} with the result._
  1. Since there still are no unit tests demonstrating that unicode paths work, the docstring shouldn't say that they do.
  1. _sibbling_ is misspelled in the `sibling` docstring; also it would be better to define the behavior of the method without using the name of the method.  Instead of _sibling_, say that the resulting path has the same directory but a different base name.
  1. There's some trailing whitespace in the patch, we try to avoid that.
  1. The `child` docstring for `path` isn't quite correct.  The `path` is the base name for the new `FilePath` to create.
  1. _String_ isn't a type :)  `str` is though.
  1. The `isabs` docstring is a little suboptimal, because `isabs` is a trick.  There is no such thing as a relative `FilePath`, they are all absolute.
  1. For `raise` epytext markup, the exception type goes before the `:`.  So, `raise OSError: ...` not `raise: OSError ...`.
  1. `FilePath.touch` doesn't actually fail silently.  If the `open('a').close()` fails, then either it's because the path couldn't be created, in which case the `utime` call will fail loudly, or because of some other problem that doesn't matter because `utime` will still succeed, or some problem which will also cause `utime` to fail, which will also be loud.
  1. For the `basename` docstring, _seperator_ is actually spelled _separator_
  1. It would be excellent for the `dirname` docstring to be parallel to the `basename` docstring, since the functionality is somewhat parallel.

These are all fairly minor points.  I'll correct them and apply the patch.  Thanks again!

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun set status to `closed`|
|-|-|

(In [31650]) Merge filepath-docstrings-5027

Author: KB1PKL, exarkun
Reviewer: exarkun
Fixes: [#5027](https://github.com/twisted/twisted/issues/5027)

Add docstrings for many of `FilePath`'s undocumented methods.
